tiledb/sm/query/reader_base.cc

```diff
@@ -74,6 +77,36 @@ ReaderBase::ReaderBase(
 /*        PROTECTED METHODS       */
 /* ****************************** */
 
+void ReaderBase::evolve_fragment_array_schemas() {
+  if (fragment_metadata_.size() <= 1) {
+    // No need to evolve
+    return;
+  }
+  for (auto i = fragment_metadata_.size() - 1; i > 0; i--) {
+    if (fragment_metadata_[i]->array_schema_name() ==
+        fragment_metadata_[i - 1]->array_schema_name()) {
+      // No need to evolve for two same schemas
+      continue;
+    }
+    auto attribute_used_names =
+        fragment_metadata_[i]->array_schema()->attribute_used_names();
+    bool need_to_build_idx = false;
+    for (auto attribute_used_name : attribute_used_names) {
+      auto attribute_current_name =
+          fragment_metadata_[i]->array_schema()->attribute_current_name(
+              attribute_used_name);
+      ArraySchema* prev_array_schema =
+          const_cast<ArraySchema*>(fragment_metadata_[i - 1]->array_schema());
+      prev_array_schema->rename_attribute(
+          attribute_used_name, attribute_current_name, false);
+      need_to_build_idx = true;
```

Comment:
Will this not be `true` in every loop iteration? I see no `if` condition anywhere.

Reply:
Yes, this bool flag is not necessary. already removed it.